src/main/java/forestry/apiculture/genetics/alleles/AlleleEffectRadioactive.java

```diff
@@ -65,43 +70,69 @@ private static IEffectData destroyEnvironment(IBeeGenome genome, IEffectData sto
         Vect offset = area.multiply(-1 / 2.0f);
         Vect posHousing = new Vect(housing.getCoordinates());
 
+        boolean isAlveary = housing instanceof AlvearyController;
+
         for (int i = 0; i < 20; i++) {
             Vect randomPos = Vect.getRandomPositionInArea(rand, area);
-            Vect posBlock = randomPos.add(posHousing);
-            posBlock = posBlock.add(offset);
-
-            if (posBlock.y <= 1 || posBlock.y >= housing.getWorld().getActualHeight()) {
-                continue;
-            }
+            Vect posBlock = randomPos.add(posHousing).add(offset);
 
-            // Don't destroy ourselves or blocks below us.
-            if (posBlock.x == posHousing.x && posBlock.z == posHousing.z && posBlock.y <= posHousing.y) {
+            // Don't destroy blocks in the protected 3x3x4 area if housing is an Alveary.
+            // Stops the bee destroying itself. Silly behaviour.
+            if (isAlveary && isInAlvearyProtectedArea(posHousing, posBlock)) {
```

Comment:
Shouldn't this be an or instead?

like:

```java
if (isAlveary || isInAlvearyProtectedArea(posHousing, posBlock))
```

Reply:
@Connor-Colenso 
Thinking of it. The condition protects the Alveary containing Radioactive Bees itself with its top slab layer; but it seems it is not protecting the slab layer on other Alvearies within the effect range.

Anyway this is not a blocking concern. This code greatly improve the existing mechanic already. It may be merged now and improved at a later stage.

Reply:
@leagris I wasn't sure how to efficiently make sure that it isn't interacting with other alvearies but I'm not that bothered, my primary concerns with this effect have been resolved and it is far more reasonable now than it was before. Thanks for the review.